### ISSUE TITLE
Chat page

### ISSUE THREAD

**Issue by sayanchowdhury** [waartaa/waartaa#220]
The chat page would contain the 
- Left panel - List of all the Networks and channels.
- Right panel - List of all the users.
- Middle panel - The chat messages. There would be textbox at the bottom for send out the text messages.
  The chat page would be having a navbar at the top for all the navigation stuffs

Also, pm page if there are visual differences from the chat page.

**Comment by mairin** [comment]
Main elements:
- Networks (connected, available to connect to)
- Channels (what networks? notifs - waiting messages, waiting pings)
- PMs (from who? what networks?)
- Users (# of users primary, specific users secondary, highlight ops)
- Me (my nick, change my nick, my status, my info, identify?)
- Commands (? eg if i'm an op, offer command via contextual buttons)
- Content - timestamps... join/leave messages.... /me messages....

If someone does a DCC request, I'm guessing we ignore it?

I'd like to have FB style link / image previews too, with some understanding that sometimes people spam in IRC with bad intentions so some rate-limiting / control on that might be nice.

Some kind of markdown style processing of text might be nice too, not sure tho.

Any other things I'm missing?

**Comment by mairin** [comment]
These are probably RFE ideas:
- ability to search logs
- ability to bookmark lines for later usage / retrieval / permanent storage / sharing
- user metadata - any way to for example show avatars via gravatar / libravatar?
- "while you were gone" summary of missed pings / notifications
- code snippet formatting
- pastebin support
- handling bots differently than people
- desktop notifications
- some kind of special handling of meetbot

Ideas specifically from roadmap:
- audio/video communication
- handling encryption
- new user wizard: "we plan to implement a wizard for new users to guide them to proper IRC channels based on their area of interest. Initially, this will be only for popular open source project. We'll keep a very simple way for the community to add content for the wizard."
- Friend finder: "A user will be allowed to follow (identified) channel nicks. If the other nick is also using Waartaa, he/she can choose to follow the follower as well. That's how two users become friends on Waartaa. Users will be able to see a list of people they follow who are online and can directly initiate chat with them irrespective of an IRC server or channel."
- keyboard shortcuts for navigation
- customizable color schemes
- In page integration with paste services like gist, pastebin, etc.
- Autocomplete IRC (server) commands, along with commands cheatlist
- UI endpoint for most IRC (server) actions
- sublime text ctrl+p like quick movement between logs (channels / pms)
- display current time / time offset for other users somehow

(roadmap: https://github.com/waartaa/waartaa_stripe_open_source_retreat/blob/master/TODOS.rst )

**Comment by sayanchowdhury** [comment]
Yes, the first list of things that you mentioned sounds good to me. Yes, we ignore DCC request for now.

We do want to let people share FB style link/image previews too but it needs a detailed analysis for the same reason you mentioned.

+1 for markdown/rst processing

**Comment by mairin** [comment]
Still working on this - this is probably the most involved screen. First idea is to have a panel on the right with tabs for different plug-in functionality like meet bot integration or a summary of links posted. Still working on it but here's a rough cut.
![chat_v1](https://cloud.githubusercontent.com/assets/799683/18091945/7a377e7c-6e98-11e6-8111-836027fadb42.png)
